tests/unit/test_urls.py

```diff
@@ -144,6 +144,26 @@ def test_dashboard_route(self):
         self.assertTrue('text/html' in rv.content_type)
         self.assertTrue(content_test)
 
+    def test_error_routes(self):
+        """Request to /error/<int> should return proper statuses.
+
+        Undefined <int> statuses return 500."""
+        rv = self.app.get('/error/404')
+        self.assertEqual(rv.status_code, 404)
+        rv = self.app.get('/error/403')
+        self.assertEqual(rv.status_code, 403)
+        rv = self.app.get('/error/4000')
+        self.assertEqual(rv.status_code, 500)
+        rv = self.app.get('/error/🤷')
+        self.assertEqual(rv.status_code, 404)
```

Comment:
@miketaylr (Pre-review question) Was there a need for creating routes for HTTP errors? HTTP errors code are supposed to inform on the specific URI you are hitting at. They don't exist with a specific URI. If we redirect to a specific route we loose the benefits of the HTTP code.

Reply:
@karlcow yeah. before these changes, we had/have a "catchall" 404, controlled by nginx. so if you want to send someone to a 404 via JS, because the API told you 404, that's easy: redirect to `https://webcompat.com/404`. It's sort of cheating, but it works (because that `/404` truly doesn't exist).

The difficulty here, AFAICT, is we don't know about the PR status until one of many HTTP requests (the API request to the `/api/issues/` endpoint) returns a 403. We can't know beforehand, unless we hard-code.

Now we need to send a 403 for a certain response. What are the options?

* Send 403, nothing happens on the page. Looks like a bug, because well, it is one. 
* Send 403, redirect to 404 page. That seems wrong.
* Send 403, redirect to a 403 page. That requires the ability to do that. That's the approach I took.

The other way to deal with this would be to display a 403 or 404 message entirely via the client. 

> If we redirect to a specific route we loose the benefits of the HTTP code.

I'm not sure I understand this point of view. If you curl `localhost:5000/api/issues/516` you get what you would expect:

```
http -I http://localhost:5000/api/issues/516
HTTP/1.0 403 FORBIDDEN
Content-Length: 58
Content-Security-Policy: default-src 'self'; object-src 'none'; connect-src 'self' https://api.github.com; font-src 'self'; img-src 'self' https://www.google-analytics.com https://*.githubusercontent.com data:; manifest-src 'self'; script-src 'self' https://www.google-analytics.com https://api.github.com; style-src 'self' 'unsafe-inline'; report-uri /csp-report
Date: Mon, 22 Jan 2018 23:04:45 GMT
Server: Werkzeug/0.12.2 Python/2.7.10
X-Content-Type-Options: nosniff
X-Frame-Options: DENY
X-XSS-Protection: 1; mode=block
cache-control: private, max-age=60, s-maxage=60
content-type: application/json
etag: W/"2d026dee32c4e3332b9f859081319ee4"

{
    "message": "Pull Requests are Forbidden.", 
    "status": 403
}
```

(it's also true if you curl `localhost:5000/error/403`, you get a 403. it feels a bit philosophical... maybe there's actually something there, but i won't let you see it).

Reply:
@miketaylr See my comment in the issue itself so we have a history on possible outcomes.
https://github.com/webcompat/webcompat.com/issues/1311#issuecomment-359689001

Reply:
Sounds like you agree with "The other way to deal with this would be to display a 403 or 404 message entirely via the client." (or rather, tell the user that they can't see that issue).

If someone wants to pick up the pieces of this PR, be my guest. Have bigger fish to fry. 🐟

Reply:
no 403, no 404. Just sending 200. but with an appropriate message. I'm happy to take over this :)

Reply:
> 403 or 404 message

Meaning, tell the user the issue is verboten. HTTP returns a 200 for HTML + other assets. The API endpoint returns a 403, which the JS reacts to and populates the template, no?

Reply:
403/404 means the URI is forbidden. We indeed return 200 for everything because of the way we are set up. Let me make the suggestion in a different PR. Do not destroy this one. :)

Reply:
(I think it makes sense that a URI that points to a PR issue should be considered forbidden by webcompat.com. 🍎 🐛

but also, not a hill i want to die on 😄 )